courses/management/commands/create_courseware.py

```diff
@@ -0,0 +1,191 @@
+            "--program",
+            type=str,
+            nargs="?",
+            help="(Course only) Add the course to the specified program (readable ID or numeric ID).",
+        )
+
+        parser.add_argument(
+            "--program-position",
+            type=str,
+            nargs="?",
+            help="(Course only) The position the course should be in the program.",
+        )
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        if kwargs["type"] == "program":
+            if Program.objects.filter(readable_id=kwargs["courseware_id"]).exists():
+                self.stderr.write(
+                    self.style.ERROR(
+                        f"Program with ID {kwargs['courseware_id']} already exists."
+                    )
+                )
+                exit(-1)
+
+            new_program = Program.objects.create(
+                readable_id=kwargs["courseware_id"],
+                title=kwargs["title"],
+                live=kwargs["live"],
+            )
+
+            self.stdout.write(
+                self.style.SUCCESS(
+                    f"Created program {new_program.id}: {new_program.title} ({new_program.readable_id})."
+                )
+            )
+        elif kwargs["type"] == "course":
+            if Course.objects.filter(readable_id=kwargs["courseware_id"]).exists():
+                self.stderr.write(
+                    self.style.ERROR(
+                        f"Course with ID {kwargs['courseware_id']} already exists."
+                    )
+                )
+                exit(-1)
+
+            program = None
+
+            if "program" in kwargs and kwargs["program"] is not None:
+                try:
+                    program = Program.objects.filter(pk=kwargs["program"])
+                except:
+                    program = Program.objects.filter(readable_id=kwargs["program"])
+
+                program = program.first()
+
+            new_course = Course.objects.create(
+                program=program,
+                title=kwargs["title"],
+                readable_id=kwargs["courseware_id"],
+                live=kwargs["live"],
+                position_in_program=kwargs["program_position"],
+            )
+
+            self.stdout.write(
+                self.style.SUCCESS(
+                    f"Created course {new_course.id}: {new_course.title} ({new_course.readable_id}) in program {new_course.program}"
+                )
+            )
+
+            if "create_run" in kwargs and kwargs["create_run"] is not None:
+                run_id = f"{new_course.readable_id}+{kwargs['create_run']}"
+
+                if CourseRun.objects.filter(
+                    run_tag=kwargs["create_run"], course=new_course
+                ).exists():
+                    self.stderr.write(
+                        self.style.ERROR(
+                            f"Course run for {new_course} with ID {kwargs['create_run']} already exists."
```

Comment:
Is it possible to hit this since we just created a the Course and so there would be no prior CourseRuns existing?

Reply:
Seems pretty unlikely indeed. I culled that in https://github.com/mitodl/mitxonline/pull/1173/commits/d5452a3851bfec808fd96f644d353462984e4843 .